component/http/cache_model.go

```diff
@@ -0,0 +1,66 @@
+package http
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+// cacheHandlerRequest is the dedicated request object for the cache handler
+type cacheHandlerRequest struct {
+	header string
+	path   string
+	query  string
+}
+
+// toCacheHandlerRequest transforms the http Request object to the cache handler request
+func toCacheHandlerRequest(req *http.Request) *cacheHandlerRequest {
+	var header string
+	if req.Header != nil {
+		header = req.Header.Get(cacheControlHeader)
+	}
+	var path string
+	var query string
+	if req.URL != nil {
+		path = req.URL.Path
+		query = req.URL.RawQuery
+	}
+	return &cacheHandlerRequest{
+		header: header,
+		path:   path,
+		query:  query,
+	}
+}
+
+// getKey generates a unique cache key based on the route path and the query parameters
+func (c *cacheHandlerRequest) getKey() string {
+	return fmt.Sprintf("%s:%s", c.path, c.query)
+}
+
+// CacheHandlerResponse is the dedicated Response object for the cache handler
+type CacheHandlerResponse struct {
+	Bytes  []byte
+	Header map[string]string
+}
+
+// CachedResponse is the struct representing an object retrieved or ready to be put into the route cache
+type CachedResponse struct {
+	Response  CacheHandlerResponse
+	LastValid int64
+	Etag      string
+	Warning   string
+	FromCache bool
+	Err       error
+}
+
+func (c *CachedResponse) encode() ([]byte, error) {
+	b, err := json.Marshal(c)
```

Comment:
as far as i understange we are caching in JSON format right?

Reply:
yes, we needed a generic encoder (for any external memory store).
I tried gob as well, but it had issues with nested objects ...
json seems to work best for generic object types (e.g. interface{} parameters)